examples/assets_smoke_test/assets_smoke_test/pure_python_assets.py

```diff
@@ -0,0 +1,41 @@
+from pandas import DataFrame
+
+from dagster import SourceAsset, TableSchema, asset
+
+raw_country_populations = SourceAsset(
+    "raw_country_populations",
+    metadata={
+        "column_schema": TableSchema.from_name_type_dict(
+            {
+                "country": "string",
+                "continent": "string",
+                "region": "string",
+                "pop2018": "int",
+                "pop2019": "int",
+                "change": "string",
+            }
+        ),
+    },
+)
+
+
+@asset
+def country_populations(raw_country_populations) -> DataFrame:
+    country_populations = raw_country_populations.copy()
```

Comment:
I guess this is necessary when outputs are passed in-memory? or is there something more to this line?

Reply:
yeah, exactly